frontend/src/framework/utils/realizationFilterTypesUtils.ts

```diff
@@ -0,0 +1,128 @@
+import {
+    DiscreteParameterValueSelection,
+    ParameterValueSelection,
+    RealizationNumberSelection,
+} from "@framework/types/realizationFilterTypes";
+
+import { isEqual } from "lodash";
+
+/**
+ * Check if parameter value selection is an array of strings.
+ */
+export function isValueSelectionAnArrayOfString(
+    valueSelection: ParameterValueSelection
+): valueSelection is readonly string[] {
+    if (Array.isArray(valueSelection) && isArrayOfStrings(valueSelection)) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Check if parameter value selection is an array of numbers.
+ */
+export function isValueSelectionAnArrayOfNumber(
+    valueSelection: ParameterValueSelection
+): valueSelection is readonly number[] {
+    if (Array.isArray(valueSelection) && isArrayOfNumbers(valueSelection)) {
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Check if array of discrete parameter values is an array of strings.
+ *
+ * DiscreteParameterValueSelection is a union type of string[] and number[].
+ */
+export function isArrayOfStrings(discreteValues: DiscreteParameterValueSelection): discreteValues is readonly string[] {
+    if (discreteValues.length === 0) {
+        return true;
+    }
+
+    // Check first element only for efficiency, as input is string[] | number[]
+    return typeof discreteValues[0] === "string";
+}
+
+/**
+ * Check if array of discrete parameter values is an array of numbers.
+ *
+ * DiscreteParameterValueSelection is a union type of string[] and number[].
+ */
+export function isArrayOfNumbers(discreteValues: DiscreteParameterValueSelection): discreteValues is readonly number[] {
+    if (discreteValues.length === 0) {
+        return true;
+    }
+
+    // Check first element only for efficiency, as input is string[] | number[]
+    return typeof discreteValues[0] === "number";
+}
```

Comment:
I think these functions are fairly general and potentially useful in other module code. Should we move it to a global `framework/utilis` file?